### DISCUSSION TITLE
Getting Started / First Steps / "Unzipped size must be smaller..." Error

### DISCUSSION THREAD

**Discussion by r2x-ch** [brefphp/bref#1019] [Support]
Hello,

I've followed your instructions for the index.php-Test-App but the file size of the app.zip file is 175.85 MB and I get this Serverless Error:

An error occurred: ApiLambdaFunction - Resource handler returned message: "Unzipped size must be smaller than 220606645 bytes (Service: Lambda, Status Code: 400, Request ID: ebc06c4a-2335-4e6a-80d0-3b74b17fcf93, Extended Request ID: null)" (RequestToken: 870432f5-7c12-60aa-1f1a-c226d272972a, HandlerErrorCode: InvalidRequest).

I didn't changed anything in index.php or serverless.yml. - How can I fix this?

My environment:
OS: Amazon Linux 2 (on VMware Workstation)
Bref Version: 1.2.14
Node Version: 16.9.1
Framework Version: 2.58.0
Plugin Version: 5.4.4
SDK Version: 4.3.0
Components Version: 3.17.0
 
Thank you!

**Comment by t-richard** [comment]
Not sure which instructions you followed but this is generally linked to a too big vendor folder or some unnecessary files likes images sent to Lambda.

You can see the sent zip in the `.serverless` folder and analyze what's inside to find the biggest files/folders. Note that the size that Lambda cares about is the **unzipped** size.

You can exclude files using this https://www.serverless.com/framework/docs/providers/aws/guide/packaging/#patterns

You can also try to reduce the vendor directory size by running `composer install --no-dev`. this will exclude dev dependencies that may account for a large portion of the directory.

Hope this helps :slightly_smiling_face:

**Comment by r2x-ch** [comment]
Thank you for your fast reply, t-richard.

I followed the Getting Started instructions from the Documentation of the bref.sh Website and created a sample web app with command `vendor/bin/bref init`. It generates a simple index.php file and the serverless.yml file.

The biggest folder is "node_modules" which should be normally excluded by pattern `- '!node_modules/**'` in the generated serverless.yml file, but for any reasons it get's not excluded.

To get at all an idea about Bref: How big the app.zip file should normally be with only a 2 kB index.php file and only bref/bref installed via Composer?

**Reply by mnapoli** [reply]
Indeed the problem might be with node_modules. Check you are running the latest Serverless version.

Usually the zip should be much much much smaller.

**Reply by t-richard** [reply]
I created a sample project with only these files and `bref/bref` as the only dependency
```
.
├── composer.json
├── composer.lock
├── index.php
├── serverless.yml
└── vendor
```

This is 5.2MB unzipped. 

For the `node_modules` that are not excluded, I can't think of a reason. Your version is the latest availble so this syntax should work. Is this `node_modules` folder at the root of the project or is it in a sub folder ?
